File: Python /two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        prevMap = {}  # Hashmap = val : index
        for i, n in enumerate(nums):
            diff = target - n
            if diff in prevMap:
                return [prevMap[diff], i]
            prevMap[n] = i


s = Solution()
print(s.twoSum([2, 7, 11, 15], 9))

File: c++/Arrays/two_sum.cpp
#include <vector>

using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    std::vector<int> result;
    for (int i = 0; i < nums.size(); i++) {
      for (int j = 0; j < nums.size() - 1; j++) {
        if ((target - nums[i]) == nums[j]) {
          result.push_back(i);
        }
      }
    }
    return result;
  };
};
